File: API/src/Api.Data/Mapping/Endereco.cs
using Api.Domain.Entities;
using Microsoft.EntityFrameworkCore;
using Microsoft.EntityFrameworkCore.Metadata.Builders;

namespace Api.Data.Mapping
{
    public class EnderecoMap : IEntityTypeConfiguration<EnderecoEntity>
    {
        public void Configure(EntityTypeBuilder<EnderecoEntity> builder)
        {
            builder.ToTable("Endereco");

            builder.HasKey(e => e.Id);

            builder.HasIndex(e => e.Cep);

            builder.HasOne(e => e.Municipio)
                    .WithMany(m => m.Enderecos);
        }
    }
}

File: Frontend/src/app/core/models/endereco.ts
export interface Uf {
  id: string;
  nome: string;
  sigla: string;
}

export interface Municipio {
  id: string;
  nome: string;
  uf: Uf;
}

export interface Endereco {
  cep: string;
  logradouro: string;
  numero: string;
  municipio: Municipio;
}
